### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte

### ISSUE THREAD

**Issue by eristocrates** [caksoylar/keymap-drawer#111]
I'm on windows 10, instillation with pipx worked fine and all that, and parsing qmk. 
I'm having an issue with unicode (maybe multibyte in general?) characters. At first i thought in my config.yaml that
  trans_legend:
    t: ��
    type: trans
was the culprit. I even tried changing that line to ▽▽ and even '' and the error (log below) persisted. It's almost like my config isn't applying it? I'll doublecheck how i'm trying to use my config, but to add to the strangeness, my actual keymap.yaml has lines like 
    - {t: ¤, type: trans}
after parsing my qmk keymp.c. very strange! currently i have to replace those strings with my own script before calling draw but I'm pretty sure this isn't normal right?

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "c:\users\reill\.local\bin\keymap.exe\__main__.py", line 7, in <module>
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\keymap_drawer\__main__.py", line 212, in main
    draw(args, config.draw_config)
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\keymap_drawer\__main__.py", line 22, in draw
    yaml_data = yaml.safe_load(args.keymap_yaml)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\__init__.py", line 79, in load
    loader = Loader(stream)
             ^^^^^^^^^^^^^^
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\loader.py", line 34, in __init__
    Reader.__init__(self, stream)
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\reader.py", line 85, in __init__
    self.determine_encoding()
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\reader.py", line 124, in determine_encoding
    self.update_raw()
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\yaml\reader.py", line 178, in update_raw
    data = self.stream.read(size)
           ^^^^^^^^^^^^^^^^^^^^^

**Comment by caksoylar** [comment]
Hey, unfortunately when I experimented in Windows I ran into similar issues. I couldn't find a set of settings to resolve it, so I documented a workaround here: https://github.com/caksoylar/keymap-drawer?tab=readme-ov-file#%E2%84%B9%EF%B8%8F-windows-command-line-issues

tl;dr is to use `-o output` instead of stdout redirecting like `>output`.

**Comment by caksoylar** [comment]
Related: #89

**Comment by eristocrates** [comment]
Yep that took care of it! thank you very much for the workaround. Shame about windows, but that's really on me for dragging my feet on leaving windows lmao

**Comment by eristocrates** [comment]
actually, i had another related issue and figured it'd be better to keep them on the same issue. Is there any way to have japanese kana drawn? i just tried and even with the -o flag and it didn't work.

**Comment by caksoylar** [comment]
Can you give an example of what you're trying? I can try to reproduce here.

**Comment by eristocrates** [comment]
just a small snippet of a single layer from keymap.yaml. keymap draw works without issue in this example
```
  - ['X', 'V', 'G', 'M', 'P', {t: '*', s: Arcane}, {t: '%', s: Arcane}, 'U', 'O', 'Y', 'B', 'Z']
  - ['J', 'K', 'S', 'N', 'D', {t: '=', s: Arcane}, {t: ':', s: Arcane}, 'A', 'E', 'I', 'H', {t: QU, s: Q}]
  - [{t: Leader, s: Core}, 'W', 'F', 'L', 'C', {t: '\', s: Arcane}, {t: '/', s: Arcane}, (, {t: ',', s: Adaptive, h: OSM Shift}, {t: '.', s: Adaptive, h: Capsword}, ), {t: Vim, s: Genuine, h: Emulation}]
  - [{t: Num Pad, h: MO Layer}, {t: R, h: CTRLR}, {t: Space, h: SML}, {t: Space, h: SMR}, {t: T, h: ALTT}]
```

however, if i attempt to replace X with た, like so
`    - ['た', 'V', 'G', 'M', 'P', {t: '*', s: Arcane}, {t: '%', s: Arcane}, 'U', 'O', 'Y', 'B', 'Z']`
    it fails with the error below. it seems kana is not recognized as a string? 
    
  Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "c:\users\reill\.local\bin\keymap.exe\__main__.py", line 7, in <module>
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\keymap_drawer\__main__.py", line 212, in main
    draw(args, config.draw_config)
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\keymap_drawer\__main__.py", line 43, in draw
    drawer = KeymapDrawer(
             ^^^^^^^^^^^^^
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\keymap_drawer\draw\draw.py", line 24, in __init__
    self.keymap = KeymapData(config=config, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\reill\AppData\Local\pipx\pipx\venvs\keymap-drawer\Lib\site-packages\pydantic\main.py", line 193, in __init__
    self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for KeymapData
layers.t
  Input should be a valid string [type=string_type, input_value={'t': 'Num Pad', 'h': 'MO Layer'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.8/v/string_type
    
    
even more strangely, on inspecting the keymap.yaml after running draw, there seems to be duplication on some {t:} keys. note the {t: {t: Num Pad below
```
     - ['た', 'V', 'G', 'M', 'P', {t: '*', s: Arcane}, {t: '%', s: Arcane}, 'U', 'O', 'Y', 'B', 'Z']
  - ['J', 'K', 'S', 'N', 'D', {t: '=', s: Arcane}, {t: ':', s: Arcane}, 'A', 'E', 'I', 'H', {t: QU, s: Q}]
  - [{t: Leader, s: Core}, 'W', 'F', 'L', 'C', {t: '\', s: Arcane}, {t: '/', s: Arcane}, (, {t: ',', s: Adaptive, h: OSM Shift}, {t: '.', s: Adaptive, h: Capsword}, ), {t: Vim, s: Genuine, h: Emulation}]
  - [{t: {t: Num Pad, h: MO Layer}, h: MO Layer}, {t: R, h: CTRLR}, {t: Space, h: SML}, {t: Space, h: SMR}, {t: T, h: ALTT}]
```

**Comment by caksoylar** [comment]
I am guessing this is an encoding issue with the input file. keymap-drawer always expects a utf-8 encoded file and Windows text editors might not default to that.

I can get this file working here, uploaded verbatim (had to change extension to txt):
[kana.txt](https://github.com/user-attachments/files/16860939/kana.txt)

When you open this in notepad (or notepad++), you can see it shows utf-8 encoding:
![image](https://github.com/user-attachments/assets/d0d0c046-e184-4fbc-8796-abfab0475aa0)

If I open a new tab in notepad and paste this in, it also seems to work because it defaults to utf-8 in my case. But I am guessing the file you are trying isn't. You can try figuring out what it is, or upload it here like I did so I can have a look. But the solution will be to save it in utf-8 encoding.

**Comment by eristocrates** [comment]
Ah, remember when i mentioned earlier that I had to use a separate script before calling draw? that was the culprit! I'm 100% taken care of now, tysm for the support o7
